src/bin/stratisd.rs

```diff
@@ -72,54 +105,103 @@ fn run() -> StratisResult<()> {
+
+    0   == Always udev fd index
+    1   == engine index if eventable
+    1/2 == Start of dbus client file descriptor(s), 1 if engine is not eventable, else 2
+    */
+    const FD_INDEX_UDEV: usize = 0;
+    const FD_INDEX_ENGINE: usize = 1;
+
 
     let mut fds = Vec::new();
 
+    fds.push(libc::pollfd {
+                 fd: udev.as_raw_fd(),
+                 revents: 0,
+                 events: libc::POLLIN,
+             });
+
     let mut eventable = engine.borrow_mut().get_eventable()?;
-    let engine_fds_end_idx = match eventable {
+
+    // The variable _dbus_client_index_start is only used when dbus support is compiled in, thus
+    // we denote the value as not needed to compile when dbus support is not included.
+    let (engine_eventable, poll_timeout, _dbus_client_index_start) = match eventable {
         Some(ref mut evt) => {
             fds.push(libc::pollfd {
                          fd: evt.get_pollable_fd(),
                          revents: 0,
                          events: libc::POLLIN,
                      });
-            1
+
+            // Don't timeout if eventable, we are event driven
+            (true, -1, FD_INDEX_ENGINE + 1)
         }
-        None => 0,
+
+        // We periodically need to timeout as we are not event driven
+        None => (false, 10000, FD_INDEX_ENGINE),
     };
 
-    // Don't timeout if eventable lets us be event-driven
-    let poll_timeout = if engine_fds_end_idx != 0 { -1 } else { 10000 };
+    #[cfg(feature="dbus_enabled")]
+    let (mut dbus_conn, mut tree, base_object_path, mut dbus_context) =
+        libstratis::dbus_api::connect(Rc::clone(&engine))?;
 
     loop {
-        // Unconditionally call evented() if engine has no eventable.
-        // This looks like a bad idea, but the only engine that has
-        // no eventable is the sim engine, and for that engine, evented()
-        // is essentially a no-op.
-        if engine_fds_end_idx == 0 {
-            engine.borrow_mut().evented()?;
+        // Process any udev block events
+        if fds[FD_INDEX_UDEV].revents != 0 {
+            loop {
+                match udev.receive_event() {
+                    Some(event) => {
+                        if let Some((devnode, device)) = handle_udev_add(&event) {
+                            let pool_uuid = engine.borrow_mut().block_evaluate(devnode, device);
+
+                            // We need to pretend that we aren't using the variable _pool_uuid so
+                            // that we can conditionally compile out the register_pool when dbus
+                            // is not enabled.
+                            if let Some(_pool_uuid) = pool_uuid {
+                                #[cfg(feature="dbus_enabled")]
+                                libstratis::dbus_api::register_pool(&mut dbus_conn,
+                                                                    &Rc::clone(&engine),
```

Comment:
The comment about the signature of the function ```register_pool``` is also a comment about this call-site. I think you should just clone here and not pass as a reference.